File: 02Fabryka/Students/2017/MarchwińskiMaciej/FactoryPattern/src/com/marchwinski/factoryexamples/stores/af/VehicleStore.java
package com.marchwinski.factoryexamples.stores.af;

import com.marchwinski.factoryexamples.vehicles.Vehicle;
import com.marchwinski.factoryexamples.vehicles.VehicleType;

public abstract class VehicleStore {
    public abstract Vehicle createVehicle(VehicleType vehicleType);

    public Vehicle orderVehicle(VehicleType type) {
        Vehicle vehicle = createVehicle(type);
        System.out.println("--- Making a " + vehicle.toString() + " ---");
        return vehicle;
    }
}

File: 02Fabryka/Students/2018/GoraPiotr/AbstracFactory/store/VehicleStore.py
from abc import ABC, abstractmethod

from Factory.AbstracFactory.vehicle.Vehicle import Vehicle


class VehicleStore(ABC):

    @abstractmethod
    def create_vehicle(self, type: str):
        pass

    def order_vehicle(self, type: str):
        vehicle: Vehicle = self.create_vehicle(type)
        vehicle.prepare()

        return vehicle
